File: Unidad 3 Variables Aleatorias/Generacion Variables Aleatorias/Composicion.py
# Descripción: Combina dos distribuciones
import random
import tkinter as tk

def generar_composicion():
    u = random.random()
    x = random.random()*2 if u<0.3 else 2 + random.random()*3
    label_resultado.config(text=f"Variable por composición: {x:.4f}")

ventana = tk.Tk()
ventana.title("Composición")
tk.Button(ventana, text="Generar", command=generar_composicion).pack()
label_resultado = tk.Label(ventana, text="Variable por composición: ?")
label_resultado.pack()
ventana.mainloop()

File: Unidad 3 Variables Aleatorias/Generacion Variables Aleatorias/Composicion.java

// Descripción: Combina dos distribuciones usando probabilidad 30%-70%
import java.util.Random;

public class Composicion {
    public static void main(String[] args) {
        Random rand = new Random();
        double u = rand.nextDouble();
        double x = (u < 0.3) ? rand.nextDouble() * 2 : 2 + rand.nextDouble() * 3;
        System.out.println("Variable por composición: " + x);
    }
}
